lib/elixir/test/elixir/kernel/overridable_test.exs

```diff
@@ -214,4 +216,20 @@ defmodule Kernel.OverridableTest do
       """
     end
   end
+
+  test "overriding private functions is deprecated and emits a warning" do
```

Comment:
I lean on putting this test here instead of in `warnings_test.exs` because this is a transient warning that will go away once we stop supporting overriding of private functions. Wdyt?